src/reducers/combined-updaters.js

```diff
@@ -170,6 +178,18 @@ export const addDataToMapUpdater = (state, {payload}) => {
 
     pick_('uiState')(apply_(toggleModalUpdater, payload_(null))),
 
+    if_(
```

Comment:
stead of directly call `pick_('uiState')` in combined updaters, you should add a `receiveMapConfigUpdater` to `ui-state-updaters`,  and import it here, call it with 
```
import {receiveMapConfigUpdater as uiStateReceiveMapConfigUpdater} from './map-style-updaters';
...
pick_('mapStyle')(apply_(styleMapConfigUpdater, payload_({config: parsedConfig, options}))),
    
pick_('uiState')(apply_(uiStateReceiveMapConfigUpdater, payload_({config: parsedConfig, options}))),

pick_('uiState')(apply_(uiStateLoadFilesSuccessUpdater, payload_(null))),
```

we have `receiveMapConfigUpdater` in each of the sub-reducers, to handle merging config into state. We should follow the pattern with uiState too

Reply:
Thanks, @heshan0131! I added `receiveMapConfigUpdater` in `ui-state-updaters`, specifically for 'mapLegend' and 'locale'. Are there any other things I should consider in this function when merging received uistate? Thanks!